src/olympia/addons/views.py

```diff
@@ -1189,6 +1189,8 @@ def get_results_count(self, results):
     def filter_queryset(self, qs):
         qs = super().filter_queryset(qs)
         guid_param = self.request.GET.get('guid')
+        if not guid_param or not amo.ADDON_GUID_PATTERN.match(guid_param):
+            raise exceptions.ParseError('Invalid guid parameter')
```

Comment:
imho, it'd be more consistent to return fallback results with a `TAAR_LITE_FALLBACK_REASON_xxx` reason, rather than raising with this API. 🤷

Reply:
I almost went that route, but decided that if the `guid` is wrong, it's really a problem with the caller, where as fallback is intended for when something goes wrong on the server.